daphne/src/testing.rs

```diff
@@ -221,18 +221,67 @@ impl DapAggregator<BearerToken> for MockAggregator {
 
     async fn get_agg_share(
         &self,
-        _task_id: &Id,
-        _batch_interval: &Interval,
+        task_id: &Id,
+        batch_interval: &Interval,
     ) -> Result<DapAggregateShare, DapError> {
-        unreachable!("not implemented");
+        let task_config = self
+            .get_task_config_for(task_id)
+            .ok_or_else(|| DapError::fatal("no task found"))?;
+
+        let mut agg_share = DapAggregateShare::default();
+
+        // Lock agg_store.
+        let mut agg_store_mutex_guard = self
+            .agg_store
+            .lock()
+            .map_err(|e| DapError::Fatal(e.to_string()))?;
+        let agg_store = agg_store_mutex_guard.deref_mut();
+
+        // Fetch aggregate shares.
+        let mut bucket_info = BucketInfo {
+            task_id: task_id.clone(),
+            window: batch_interval.start,
+        };
+        // TODO(nakatsuka-y) Should the batch window be inclusive?
+        while bucket_info.window < batch_interval.end() {
+            if let Some(delta) = agg_store.get_mut(&bucket_info) {
+                agg_share.merge(delta.clone())?;
+            }
+
+            bucket_info.window += task_config.min_batch_duration;
+        }
```

Comment:
This is extremely inefficient if the keys for agg_store are sparse. What about iterate over the map? Something like
```rust
for (inner_bucket_Info, agg_share_deltaj) in agg_store.iter() {
      if bucket_info.task_id == inner_bucket_info.task_id &&
         bucket_Info.start <= inner_bucket_info.start &&
         bucket_info.start + bucket_Info.duration > inner_bucket_Info.start {
               // Add delta into `agg_share`
      }
}
```

Reply:
Agreed. I've changed the while loop to iterate over the records in the HashMap.